### ISSUE TITLE
sql: settings are flaky in tests

### ISSUE THREAD

**Issue by arjunravinarayan** [cockroachdb/cockroach#15448]
Settings are currently racing in logic tests when swapped back and forth, e.g.
```
test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic_default_set546458208
test_log_scope.go:62: use -show-logs to present logs inline
logic_test.go:1625: 
    testdata/logic_test/set:235: 
    expected:
        2
        
    but found (query options: "") :
        1  
    
logic_test.go:1655: testdata/logic_test/set:240: too many errors encountered, skipping the rest of the input
```

They are not being tested often enough to reliably cause races, until the introduction of `EnumSetting`. cc @dt.

**Comment by dt** [comment]
short-term solution: we should not be calling `SET CLUSTER SETTING` in a logic test (which defaults to parallel) but rather in a normal, non-parallel unit test (I'd probably move these to settingsworker_test.go).

**Comment by knz** [comment]
Note: i'm disabling the tests on master until this is fixed, see #15458

**Comment by arjunravinarayan** [comment]
Thanks for doing that @knz!

**Comment by arjunravinarayan** [comment]
This was closed by #15461.
